bundles/org.openhab.binding.linky/src/main/java/org/openhab/binding/linky/internal/handler/ThingLinkyRemoteHandler.java

```diff
@@ -0,0 +1,820 @@
+                REFRESH_MINUTE_OF_DAY, () -> {
+                    LocalDate today = LocalDate.now();
+                    MeterReading meterReading = getPowerData(today.minusDays(1095), today);
+                    meterReading = getMeterReadingAfterChecks(meterReading);
+                    if (meterReading != null) {
+                        logData(meterReading.baseValue, "Day (peak)", DateTimeFormatter.ISO_LOCAL_DATE, Target.ALL);
+                    }
+                    return meterReading;
+                });
+
+        // Comsuption Load Curve
+        this.loadCurveConsumption = new ExpiringDayCache<>("loadCurveConsumption", REFRESH_HOUR_OF_DAY,
+                REFRESH_MINUTE_OF_DAY, () -> {
+                    LocalDate today = LocalDate.now();
+                    MeterReading meterReading = getLoadCurveConsumption(today.minusDays(6), today);
+                    meterReading = getMeterReadingAfterChecks(meterReading);
+                    if (meterReading != null) {
+                        logData(meterReading.baseValue, "Day (peak)", DateTimeFormatter.ISO_LOCAL_DATE, Target.ALL);
+                    }
+                    return meterReading;
+                });
+    }
+
+    @Override
+    public synchronized void initialize() {
+        logger.debug("Initializing Linky handler for {}", config.prmId);
+
+        // reread config to update timezone field
+        config = getConfigAs(LinkyThingRemoteConfiguration.class);
+
+        Bridge bridge = getBridge();
+        if (bridge == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_UNINITIALIZED, "@text/offline.missing-bridge");
+            return;
+        }
+
+        if (bridge.getHandler() instanceof BridgeRemoteBaseHandler bridgeHandler) {
+            enedisApi = bridgeHandler.getEnedisApi();
+            divider = bridgeHandler.getDivider();
+
+            updateStatus(ThingStatus.UNKNOWN);
+
+            if (config.seemsValid()) {
+                if (config.timezone.isBlank()) {
+                    zoneId = this.timeZoneProvider.getTimeZone();
+                } else {
+                    zoneId = ZoneId.of(config.timezone);
+                }
+
+                if (bridgeHandler instanceof BridgeRemoteApiHandler && config.prmId.isBlank()) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                            "@text/offline.config-error-mandatory-settings");
+                    return;
+                }
+
+                if (!config.prmId.isBlank()) {
+                    bridgeHandler.registerNewPrmId(config.prmId);
+                }
+                pollingJob = scheduler.schedule(this::pollingCode, 5, TimeUnit.SECONDS);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "@text/offline.config-error-mandatory-settings");
+
+                return;
```

Comment:
This return can be removed.

You still do not understand what I ask. I want to cover the case where the thing handler is not yet initialized.
So what I would like to see is something like that:
```
        if (bridge.getHandler() instanceof BridgeRemoteBaseHandler bridgeHandler) {
            enedisApi = bridgeHandler.getEnedisApi();
            ....
        } else {
            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_UNINITIALIZED);
        }
```